### ISSUE TITLE
Fire and Forget Request (without Response)

### ISSUE THREAD

**Issue by AhmedNourJamalElDin** [visionmedia/superagent#1530]
Hello,

I was wondering if I can make a request without waiting any response?

I've set the **response** timeout to 1 millisecond, but the server didn't receive my request.
I had to increase it much more to get it working but I'm not interested in the response at all, I just want to make sure I got connected to the other server and the server got my request successfully.

P.S. I have no control on the server and I want to make sure my request has reached the endpoint.

**Comment by SystemParadox** [comment]
If you want to make sure the server got your request then that's a response. If you don't want to do anything with the response then just ignore it, but there's no need to do anything special here.

**Comment by AhmedNourJamalElDin** [comment]
aha, but I can't wait the response and I can't make it async request. 

TBH I use AWS Lambda so more time means more money and I'm not interested in the response.

**Comment by SystemParadox** [comment]
I think you completely misunderstand how Javascript works. What you're asking for doesn't make sense. Waiting for the response does not cost CPU time and it doesn't block the rest of your code from running. The request is async, whether you like it or not (XMLHttpRequest has an option to be synchronous, but it is deprecated and should *NEVER* be used, ever).

**DO NOT** set the response timeout low to try and achieve this. That is absolutely not what you want and will cause your app to break randomly if the network latency to the server varies at all.

If you want to throw an error if you got something other than 200 response, do this:

```
superagent.post('/foobar').then(res => {
    if (res.status != 200) {
        throw new Error('Failed');
    }
});
```

If you don't even care about the response code, you'll get an error anyway if superagent fails to connect or get any response:

```
superagent.post('/foobar');
```

In both cases, this is a promise rejection, not a synchronous error. Unless you are using `async...await` then you can't catch this with `try...catch`, you have to use `.catch()` or the `unhandledPromiseRejection` event.

I would recommend reading up about promises and callbacks in Javascript/node to understand better how this all works.

**Comment by niftylettuce** [comment]
Thanks for chiming in @SystemParadox 👍 	Appreciate your help with this

**Comment by AhmedNourJamalElDin** [comment]
Well, this won't work in AWS lambda. thanks anyway @SystemParadox

**Comment by SystemParadox** [comment]
@AhmedNourJamalElDin, sorry no that is not correct. You misunderstand how this all works, and what you are asking for is both impossible and pointless.

- If AWS lambda charges for CPU time, then this is all completely irrelevant and you should just send the request and not care. You do not save anything by setting a short timeout. It doesn't cost CPU to wait for a response.
- If for some reason AWS lambda charges for real time (I find this unlikely), then you could set a timeout to limit the maximum time for the request - but any request that takes longer than this timeout will be aborted, which in most cases means the request will not reach the server. This timeout should be **WELL ABOVE** the usual response time - several seconds at least. If for example you set the timeout to 10 seconds, if most requests finish within 2 seconds then you'll be done within 2 seconds, and any request that takes longer than 10 seconds will be aborted.

There is no concept of a fire-and-forget request in HTTP. The only way you know that the server has actually recieved your request is when it sends the response. If you abort before that then you're almost certainly aborting the request before it reached the server. This is nothing to do with Node or Javascript, it's the way HTTP works. It is not safe for your app to exit or close the connection until you have received the response. Even if your request has actually reached the server, if you close the connection before the server has processed it then the server will see that the connection has been dropped and it won't bother processing your request.

**Comment by ebisbe** [comment]
@SystemParadox I came here because I was searching for something similar to what @AhmedNourJamalElDin was searching ( also in the AWS Lambda space ). 
Just to clarify, despite this being 4 years old. Lambda charges you by **CPU usage** and **time spent during an invocation**.  That has always been their pricing model.

Now to my real use case with actual data of why I'm searching for something that could be a fire&forget call. I've set up a timeout of 2s on each request ( because I need an actual value ) and in some cases the timeout triggers. I have a try catch that retries the request and the second time I get an error telling me that the photo is already in that group, meaning that the request succeed despited having to time it out.
